lib/broccoli/angular-broccoli-less.js

```diff
@@ -0,0 +1,69 @@
+/* jshint node: true, esversion: 6 */
+'use strict';
+
+try {
+  let less;
+
+  if (process.platform === 'win32') {
+    require.resolve(`${process.env.PWD}/node_modules/less`);
+    less = require(`${process.env.PWD}/node_modules/less`);
+  } else {
+    process.env.NODE_PATH += `:${process.env.PWD}/node_modules`;
+    require('module').Module._initPaths();
+    require.resolve('less');
+    less = require('less');
+  }
+
+  const Plugin  = require('broccoli-caching-writer');
+  const fs      = require('fs');
+  const fse     = require('fs-extra');
+  const path    = require('path');
+  const Funnel  = require('broccoli-funnel');
+
+  class LESSPlugin extends Plugin {
+    constructor(inputNodes, options) {
+      super(inputNodes, {});
+
+      options = options || {};
+      Plugin.call(this, inputNodes, {
+        cacheInclude: [/(.*?).less$/]
+      });
+      this.options = options;
+      this.fileRegistry = [];
+    }
+
+    build() {
+      let entries = this.listEntries();
+      let rootFileNames = entries.map(e => {
+        return path.resolve(e.basePath, e.relativePath);
+      });
+
+      return Promise.all(rootFileNames.map(fileName => {
+        return this.compile(fileName, this.inputPaths[0], this.outputPath);
+      }));
```

Comment:
Again the function body could be:

```js
return Promise.all(this.listEntries().map(e => {
  const fileName = path.resolve(e.basePath, e.relativePath);
  return this.compile(fileName, this.inputPaths[0], this.outputPath);
}));
```

Or even:

```js
return Promise.all(this.listEntries().map(e => this.compile(path.resolve(e.basePath, e.relativePath), this.inputPaths[0], this.outputPath)));
```

:P _(I am not seriously proposing the latter)_